docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+       metadata:
+         namespace: metalk8s-monitoring
+         name: <example-name>
+       data:
+         connector.ldap.host: 'ldap.example.com:636'
+         connector.ldap.startTLS: true
+         connector.ldap.bindDN: ''
+         connector.ldap.bindPW: ''
+         connector.ldap.rootCA: ''
+
+         connector.ldap.userSearch.baseDN: ''
+         connector.ldap.userSearch.filter: '(objectClass=inetOrgPerson)'
+         connector.ldap.userSearch.username: ''
+         connector.ldap.userSearch.idAttr: 'DN'
+         connector.ldap.userSearch.emailAttr: ''
+         connector.ldap.userSearch.nameAttr: 'cn'
+
+         connector.ldap.userSearch.groupSearch.baseDN: ''
+         connector.ldap.userSearch.groupSearch.filter: '(objectClass=groupOfUniqueNames)'
+         connector.ldap.userSearch.groupSearch.userAttr: 'DN'
+         connector.ldap.userSearch.groupSearch.groupAttr: 'uniqueMember'
+         connector.ldap.userSearch.groupSearch.nameAttr: 'cn'
+
+         # setting this to false should disable the use of static users
+         # for users who own no LDAP or AD, add configs allowing provisioning
+         # more static users or changing the password for the default user.
+         # TBD
+         connector.staticUser.enabled: true
+
+**Use case 3:**
+
+Configure and store Alertmanager notification receivers so that alerts can
+be dispatched to the customers choice of notifier(Slack, email, hipchat).
+
+   .. code-block:: yaml
+
+       apiVersion: v1
+       kind: ConfigMap
+       metadata:
+         namespace: metalk8s-monitoring
+         name: <example-name>
+       data:
+
+        channel.smtp.smtp_smarthost: 'smtp.gmail.com:587'
+        channel.smtp.smtp_from: 'foo@bar.com'
+        channel.smtp.smtp_auth_username: 'foo@bar.com'
+        channel.smtp.smtp_auth_password: 'barfoo'
+
+        channel.slack.slack_api_url: 'https://hooks.slack.com/services/abc123'
+
+        channel.hipchat.hipchat_auth_token: '1234556789'
+        channel.hipchat.hipchat_url: 'https://hipchat.foobar.org/'
+
+        # route.group_wait: '30s'
+        # route.repeat_interval: '15m'
+        # route.group_interval: '5m'
+        route.receiver: 'default'
+
+        routes.match.severity: 'email'
+        routes.receiver: 'default'
+        routes.match.severity: 'slack'
+        routes.match.receiver: 'slack_alert'
+
+        receivers.name: 'default'
+        receivers.email_configs.to: 'foo@bar.com'
+
+        receivers.name: 'slack_alert'
+        receivers.slack_configs.channel: '#devops'
+        receivers.slack_configs.send_resolved: 'true'
+
+Iteration 1
+~~~~~~~~~~~
+
+- Define and deploy new namespaced bounded ConfigMap KV stores that will hold
+  user configurations as listed in the requirements
+- Template and render Deployment and Pod manifests that will make use of
+  this user customizable configurations using default pillar values
+- Provide a CLI tool(Bash script) for changing any of the user configurations:
```

Comment:
Why Bash instead of Python?
I thought we were moving away from Shell script.

Reply:
Very true @slaperche-scality 
I think we have an entire EPIC to handle this migration and it is debatable if we should wait or have something working(shell scripts) while waiting.

Maybe, after all, we do not even have to handle this point within this EPIC

Reply:
The other EPIC is about migrating existing shellscript.

Meanwhile, for new ones that we creates from scratch, it could be a good idea to use Python from the beginning (except if it's really more costly than Bash for some reasons): would be less work for the other EPIC, while adding no extra-work to the current one (anyway you have to write a new script)

Reply:
In my opinion, wait for the Python CLI epic (#2163) to have the foundations set in place before implementing a CLI for this. In any case, the CLI / UI implementation of those customizations is less urgent than having a framework to persist said customizations over upgrade/downgrade/... IMO, CLI / UI shouldn't be part of the first iteration.